includes/discovery/route/routeros.inc.php

```diff
@@ -19,56 +19,65 @@
  * @package    LibreNMS
  * @link       https://www.librenms.org
  *
- * @copyright  2022 Peca Nesovanovic
+ * @copyright  2025 Peca Nesovanovic
  *
  * @author     Peca Nesovanovic <peca.nesovanovic@sattrakt.com>
  */
 
+use App\Facades\PortCache;
+use Illuminate\Support\Facades\Log;
+use LibreNMS\Exceptions\InvalidIpException;
 use LibreNMS\Util\IPv6;
 
-$oids = SnmpQuery::walk('IPV6-MIB::ipv6RouteTable')->table(1);
+$oids = SnmpQuery::cache()->hideMib()->walk('IPV6-MIB::ipv6RouteTable')->table(3);
 
-foreach ($oids as $dst => $data) {
-    $PfxLen = array_key_first($data['IPV6-MIB::ipv6RouteIfIndex']);
-    $RouteIndex = array_key_first($data['IPV6-MIB::ipv6RouteIfIndex'][$PfxLen]);
+foreach ($oids as $dst => $tdata) {
+    $pfxLen = key($tdata);
+    $tdata = array_shift($tdata);
 
-    //route destination
-    $ipv6dst = new IPv6($dst);
-    $dst_uncompressed = $ipv6dst->uncompressed();
+    foreach ($tdata as $timestamp => $data) {
+        try {
+            //route destination
+            $ipv6dst = IPv6::fromHexString($dst);
+            $dst_uncompressed = $ipv6dst->uncompressed();
 
-    //next hop
-    $ipv6hop = new IPv6($data['IPV6-MIB::ipv6RouteNextHop'][$PfxLen][$RouteIndex]);
-    $hop_uncompressed = $ipv6hop->uncompressed();
+            //next hop
+            $ipv6hop = IPv6::fromHexString($data['ipv6RouteNextHop']);
+            $hop_uncompressed = $ipv6hop->uncompressed();
 
-    //portId from ifIndex
-    $ifIndex = $data['IPV6-MIB::ipv6RouteIfIndex'][$PfxLen][$RouteIndex];
-    $portId = \App\Facades\PortCache::getIdFromIfIndex($ifIndex, $device['device_id']);
+            //portId from ifIndex
+            $ifIndex = $data['ipv6RouteIfIndex'];
+            $portId = PortCache::getIdFromIfIndex($ifIndex, $device['device_id']);
 
-    //populate array with data
-    unset($entryClean);
-    $entryClean['updated_at'] = $update_timestamp;
-    $entryClean['device_id'] = $device['device_id'];
-    $entryClean['port_id'] = $portId;
-    $entryClean['context_name'] = '';
-    $entryClean['inetCidrRouteIfIndex'] = $ifIndex;
-    $entryClean['inetCidrRouteType'] = $data['IPV6-MIB::ipv6RouteType'][$PfxLen][$RouteIndex];
```

Comment:
I know this is how the old code did it, but do not create an array like this... can cause side effects and that is why "unset()" is needed above.  do this instead:
$entryClean = [
// values
];